lang/jython/Portfile

```diff
@@ -18,13 +18,10 @@ long_description    \
     language Python written in 100% Pure Java, and seamlessly integrated with \
     the Java platform. It thus allows you to run Python on any Java platform.
 
-homepage            http://www.jython.org/
+homepage            https://www.jython.org/
 
-# Jython 2.7.1 is only compatible with Java 7/8, and is known to be
-# incompatible with Java 9+ (this will likely be resolved by Jython 2.7.2).
-# Since the Java portgroup doesn't support specifying a maximum Java version
-# but supports specifying an exact version, only use Java 8 for now.
-java.version        1.8
+java.version        1.7+
```

Comment:
I had the following issue with this change to an "or later" version specification:

- I have openjdk8 and openjdk13 installed
- When installing jython with `-installer`, the `1.7+` version spec resolved to openjdk13
- Jython was thus compiled with openjdk13
- However my `JAVA_HOME` is generally set to Java 1.8; when launching Jython I get this error:

    ```
    Exception in thread "main" java.lang.NoSuchMethodError: java.nio.ByteBuffer.clear()Ljava/nio/ByteBuffer;
        at org.python.core.io.BufferedReader.clear(BufferedReader.java:147)
        at org.python.core.io.BufferedReader.<init>(BufferedReader.java:27)
        at org.python.core.PyFile.createBuffer(PyFile.java:227)
        at org.python.core.PyFile.file___init__(PyFile.java:185)
        at org.python.core.PyFile.file___init__(PyFile.java:178)
        at org.python.core.PyFile.<init>(PyFile.java:101)
        at org.python.core.PySystemState.<init>(PySystemState.java:236)
        at org.python.core.PySystemState.doInitialize(PySystemState.java:1222)
        at org.python.core.PySystemState.initialize(PySystemState.java:1130)
        at org.python.core.PySystemState.initialize(PySystemState.java:1085)
        at org.python.core.PySystemState.initialize(PySystemState.java:1080)
        at org.python.core.PySystemState.initialize(PySystemState.java:1075)
        at org.python.util.jython.main(jython.java:533)
    ```

The cause is this: https://jira.mongodb.org/browse/JAVA-2559

The best solution to me seems to be to require precisely Java 1.8, as the result can then be executed by any version.

This solves the Apache Ant issue as well (I do think that the world has moved past Java 7 and Ant 1.9, and we shouldn't bother supporting building with these).